update gateway api docs to v1beta1 (#1321)
fluxcd/flagger: docs/gitbook/tutorials/gatewayapi-progressive-delivery.md

## Prerequisites

Flagger requires a Kubernetes cluster **v1.19** or newer and any mesh/ingress that implements the `v1beta1` version of Gateway API. We'll be using Contour for the sake of this tutorial, but you can use any other implementation.

> > Note: Flagger supports `v1alpha2` version of Gateway API but its deprecated and support will be dropped in a future release soon.

Reviewer: ```suggestion
> Note: Flagger supports `v1alpha2` version of Gateway API, but the alpha version has been deprecated and support will be dropped in a future release.
```
